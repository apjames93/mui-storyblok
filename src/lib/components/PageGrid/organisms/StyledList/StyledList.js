/* eslint-disable max-len */
import React, { lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import { Column } from '@mui-treasury/components/flex';
import { validComponents } from 'lib/utils/customProps';
import { renderComponentsWithBridge } from 'lib/utils/customComponents';

/* istanbul ignore next */
const MusicListItem = lazy(() => import('./components/MusicListItem/MusicListItem'));
/* istanbul ignore next */
const ChatzListItem = lazy(() => import('./components/ChatzListItem/ChatzListItem'));
/* istanbul ignore next */
const TrendyListItem = lazy(() => import('./components/TrendyListItem/TrendyListItem'));

const components = {
  MusicListItem,
  ChatzListItem,
  TrendyListItem,
};

const StyledList = ({
  width,
  content,
  storyblokClass,
  dataBlokC,
  dataBlokUid,
}) => (
  <Column
    gap={2}
    width={width}
    className={storyblokClass}
    data-blok-c={dataBlokC}
    data-blok-uid={dataBlokUid}
  >
    {content.map((component, key) => (
      <Suspense fallback={<></>}>
        {renderComponentsWithBridge(components, component, key)}
      </Suspense>
    ))}
  </Column>
);

export default StyledList;

StyledList.propTypes = {

  /** width of list */
  width: PropTypes.string,

  /** MuiListItem */
  content(props, propName, componentName) {
    const comps = ['ListItem'];
    return validComponents(props, propName, componentName, comps);
  },
  /** storyblok prop for when in editor to allow click bridge */
  dataBlokC: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  dataBlokUid: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  storyblokClass: PropTypes.string,
};

StyledList.defaultProps = {
  width: '100%',
  content: [],
  dataBlokC: '',
  dataBlokUid: '',
  storyblokClass: '',
};
