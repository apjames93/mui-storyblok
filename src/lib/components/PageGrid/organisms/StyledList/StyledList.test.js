import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import StyledList from './StyledList';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "StyledList", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    content: [{
      _editable: '<!--#storyblok#{"name": "MusicListItem", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MusicListItem',
      title: 'title',
      subtitle: 'subtitle',
      src: 'cool.png',
    }],
  };
  const comp = shallow(<StyledList {...props} />);
  return { comp, props };
}

describe('<StyledList />', () => {
  it('renders StyledList', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<StyledList {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
