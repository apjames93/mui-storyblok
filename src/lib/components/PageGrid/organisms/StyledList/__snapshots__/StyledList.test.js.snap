// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledList /> snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-1 FlexItem FlexColumn"
/>
`;
